main/src/main/scala/sbt/internal/server/BuildServerReporter.scala

```diff
@@ -116,14 +129,17 @@ final class BuildServerReporterImpl(
         exchange.notifyEvent("build/publishDiagnostics", params)
       }
     }
-  }
 
   override def sendFailureReport(sources: Array[VirtualFile]): Unit = {
     for {
       source <- sources
       filePath <- toSafePath(source)
     } {
-      val diagnostics = problemsByFile.getOrElse(filePath, Vector())
+      val diagnostics = problemsByFile.getOrElse(filePath, Vector.empty)
+
+      // mark file as file with problems
+      hasAnyProblems.add(filePath)
```

Comment:
Here I guess we should check if `diagnostics` is empty or not
```suggestion
      if (diagnostics.nonEmpty)
        hasAnyProblems.add(filePath)
```